### PR TITLE
no aux::settings in storage_interface

### PR DESCRIPTION
**Pull request by aldenml**
I think this PR is more easy to read and safer. I didn't mix the general "flags" with the "open flags" and in fact, I discovered serious bugs mixing flags in my previous PR.

### CODE REVIEW

**Comment by aldenml** [comment]
very quiet here :) (I don't mind closing it if you don't like it)

File: include/libtorrent/storage.hpp [hunk 1]
```diff
@@ -247,9 +243,9 @@ namespace libtorrent
 		// error. If there's an error, the ``storage_error`` must be filled out
 		// to represent the error that occurred.
 		virtual int readv(span<file::iovec_t const> bufs
-			, int piece, int offset, int flags, storage_error& ec) = 0;
+			, int piece, int offset, int flags, int open_flags, storage_error& ec) = 0;
```
Comment (arvidn) [inline include/libtorrent/storage.hpp]: the thing that worries me a bit here is how all the ``int`` arguments aren't typed. It makes it easy to get the order wrong. obviously this is not a new problem, it's just compounding.

File: include/libtorrent/storage.hpp [hunk 2]
```diff
@@ -594,11 +584,14 @@ namespace libtorrent
 		// when wrong in the disk access
 		int check_fastresume(add_torrent_params const& rd
 			, std::vector<std::string> const& links
+			, aux::session_settings const& settings
```
Comment (arvidn) [inline include/libtorrent/storage.hpp]: how come the settings are necessary here?

File: src/storage.cpp [hunk 3]
```diff
@@ -1498,13 +1490,13 @@ namespace libtorrent
 	}
 #endif
 
-	int piece_manager::check_no_fastresume(storage_error& ec)
+	int piece_manager::check_no_fastresume(aux::session_settings const& settings
+		, int const open_flags, storage_error& ec)
 	{
-		bool has_files = false;
-		if (!m_storage->settings().get_bool(settings_pack::no_recheck_incomplete_resume))
+		if (!settings.get_bool(settings_pack::no_recheck_incomplete_resume))
```
Comment (arvidn) [inline src/storage.cpp]: ah, for the ``no_recheck_incomplete_resume``

**Comment by arvidn** [comment]
looks reasonable. I'm not happy with all those ``int``s long term really, but it's possible the ``storage_interface`` may get collapsed anyway.

**Comment by aldenml** [comment]
Ok, if `storage_interface` is schedule to a heavy refactor, I prefer not doing this change, this it's just a marginal improvement, at the expense of the int flags explosion (with a lot of API downsides). In the swig side, I'm able to filter out that field and method without problems. I'm closing this, but keeping the branch in my fork.

**Pull request closed without merging**